src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/PremisesEntity.kt

```diff
@@ -174,8 +174,29 @@ where
 
 @Repository
 interface ApprovedPremisesRepository : JpaRepository<ApprovedPremisesEntity, UUID> {
-  @Query("SELECT p as premises FROM ApprovedPremisesEntity p WHERE :gender IS NULL OR p.gender = :gender")
-  fun findForSummaries(gender: ApprovedPremisesGender?): List<ApprovedPremisesEntity>
+  @Query(
+    """
+        SELECT
+          new uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.ApprovedPremisesBasicSummary(
+              p.id,
+              p.name,
+              apArea.id,
+              apArea.name,
+              CAST(COUNT(b) as int)
+              )
+        FROM 
+          ApprovedPremisesEntity p
+          LEFT JOIN p.rooms r 
+          LEFT JOIN r.beds b on (b.endDate IS NULL OR b.endDate > CURRENT_DATE) 
+          LEFT JOIN p.probationRegion region
+          LEFT JOIN region.apArea apArea
+        WHERE 
+          (:gender IS NULL OR p.gender = :gender)
+          AND(cast(:apAreaId as text) IS NULL OR apArea.id = :apAreaId) 
+          GROUP BY p.id, p.name, apArea.id, apArea.name 
+      """,
```

Comment:
A couple of things on this query we might want to consider (if have not already!):
1. do we need to add an index onto `beds.endDate` column as that drives the counts and might be slow?
2. do we need to add an an index to the `approved_premises.gender` column also to speed up this bit in the where clause `p.gender = :gender`?

Reply:
Although this query is basically a cut-down copy of another which already exists, I'm going to take an action to look into the performance of the query in the two areas you highlight.